crates/sgx_validation/src/config.rs

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use cryptography::digest::Digest;
+use serde::{Deserialize, Deserializer, Serialize};
+use sgx::parameters::Features;
+
+#[derive(Clone, Deserialize, Debug, Serialize)]
+pub enum SgxFeatures {
+    CET,
+    Debug,
+    EIntKey,
+    KSS,
+    ProvisioningKey,
+}
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ConfigIntermediate {
+    /// Values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub enarx_signer: Vec<Digest>,
+
+    /// Values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    pub enarx_hash: Vec<Digest>,
+
+    /// Values for `features`.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_features")]
+    pub features: u64,
+
+    /// Minimum value for `isv_svn`.
+    pub enclave_security_version: Option<u16>,
+
+    /// Value for `isv_prodid`, do not allow versions below this.
+    pub enclave_product_id: Option<u16>,
+}
+
+fn from_features<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Vec<SgxFeatures> = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = Features::empty();
+
+    flags |= Features::INIT; // Must be set
+    flags |= Features::MODE64BIT; // Required by Enarx
```

Comment:
Same as above. If Enarx does not require this tomorrow, how would anyone looking at this know?

Reply:
That won't ever happen. `Features::INIT` is required by SGX, and I'm sure we're not going to stop using 64-bit in favour of 32-bit.

Reply:
As I said above, either document here or (better) add a reference to permanent location where it is documented.
The existing comments are useless, they do not communicate anything to the reader. In fact, you seem to suggest that `64BIT` is not *required*, but *chosen* by Enarx. Again, that should be documented

Reply:
I believe the comments I added are satisfactory as per our call today.